tests/v1beta3/authconfig.yaml

```diff
@@ -8,13 +8,9 @@ spec:
 
   patterns:
     admin-path:
-    - selector: context.request.http.path
-      operator: matches
-      value: ^/admin(/.*)?$
+    - predicate: request.http.path.matches("^/admin(/.*)?$")
```

Comment:
Out of curiosity, are these changes still WIP for this file? 🤔 
Just wondering cause running `make e2e` is erroring due to invalid AuthConfig with this since it's not expecting `predicate` currently:
- https://github.com/Kuadrant/authorino/blob/3925889e8753a51592b169535c1e5bc36a630afc/api/v1beta3/auth_config_types.go#L118
- https://github.com/Kuadrant/authorino/blob/3925889e8753a51592b169535c1e5bc36a630afc/api/v1beta3/auth_config_types.go#L152-L165
```log
Error from server (BadRequest): error when creating "/Users/chfan/dev/authorino/tests/v1beta3/authconfig.yaml": AuthConfig in version "v1beta3" cannot be handled as a AuthConfig: strict decoding error: unknown field "spec.patterns.admin-path[0].predicate", unknown field "spec.patterns.resource-path[0].predicate"
waiting authconfig readyError from server (NotFound): authconfigs.authorino.kuadrant.io "e2e-test" not found
.Error from server (NotFound): authconfigs.authorino.kuadrant.io "e2e-test" not found
```
And also, failing on the `oneOf` validations such as:
- https://github.com/Kuadrant/authorino/blob/3925889e8753a51592b169535c1e5bc36a630afc/install/crd/patches/oneof_in_authconfigs.yaml#L305-L321

```log
The AuthConfig "e2e-test" is invalid: 
* <nil>: Invalid value: "": "spec.response.success.headers.wristband.when[0]" must validate one and only one schema (oneOf). Found none valid
* spec.response.success.headers.wristband.when[0].patternRef: Required value
* <nil>: Invalid value: "": "spec.authentication.anonymous.when[0]" must validate one and only one schema (oneOf). Found none valid
* spec.authentication.anonymous.when[0].patternRef: Required value
* <nil>: Invalid value: "": "spec.authentication.anonymous.when[1]" must validate one and only one schema (oneOf). Found none valid
* spec.authentication.anonymous.when[1].patternRef: Required value
* <nil>: Invalid value: "": "spec.authorization.admin-kubernetes-rbac.when[1]" must validate one and only one schema (oneOf). Found none valid
* spec.authorization.admin-kubernetes-rbac.when[1].patternRef: Required value
* <nil>: Invalid value: "": "spec.authorization.admin-jwt-rbac.when[1]" must validate one and only one schema (oneOf). Found none valid
* spec.authorization.admin-jwt-rbac.when[1].patternRef: Required value
* <nil>: Invalid value: "": "spec.authorization.admin-jwt-rbac.patternMatching.patterns[0]" must validate one and only one schema (oneOf). Found none valid
* spec.authorization.admin-jwt-rbac.patternMatching.patterns[0].patternRef: Required value
waiting authconfig readyError from server (NotFound): authconfigs.authorino.kuadrant.io "e2e-test" not found
```

Reply:
Darn, looks like I screwed up my rebasing then 🤦

Reply:
Actually... no! I made this fail on purpose, I need @guicassolato 's opinion here... cause adding CEL there is either recursive _or_ a breaking change, or needs another `NamedCelExpression` mechanism, or...

Reply:
So, I'm thinking of adding a `celFunctions` at the same level as 

```go
	NamedPatterns map[string]PatternExpressions `json:"patterns,omitempty"`
```

Same "pattern", but `map[string]CelExpression` that then register the key as a fn() in cel. So that:

```yaml
celFunctions:
  userGroup: request.headers['someHeader']
```

become usable in a cel expression: `userGroup().startWith("admin")`.

If we think that makes sense, I'd add this in a subsequent PR tho and "just get rid of `NamedPattern`s equivalence for Cel for now... wdyt?